sdk/openai/azure-ai-openai/src/main/java/com/azure/ai/openai/models/EmbeddingItem.java

```diff
@@ -64,4 +65,32 @@ public List<Double> getEmbedding() {
     public int getPromptIndex() {
         return this.promptIndex;
     }
+
+    /**
+     * Creates an instance of EmbeddingItem class.
+     *
+     * @param embedding the embedding value to set.
+     * @param promptIndex the promptIndex value to set.
+     */
+    @JsonCreator
+    private EmbeddingItem(@JsonProperty(value = "embedding") BinaryData embedding,
+        @JsonProperty(value = "index") int promptIndex) {
+        this.embedding = embedding;
+        this.promptIndex = promptIndex;
+        embeddingFloatList = convertBase64ToFloatList(embedding.toString());
```

Comment:
Since we are calling `embedding.toString()` here, we should not call this again when `getEmbeddingAsString()` method is called. We can have another property `embeddingBase64` to hold this value.